app/frontend/components/pages/sendAda/sendAdaPage.tsx

```diff
@@ -222,25 +228,42 @@ const SendAdaPage = ({
           fieldValue,
           token: {
             policyId: dropdownAssetItem.policyId,
-            assetName: dropdownAssetItem.assetNameHex,
-            quantity: parseFloat(fieldValue),
+            assetName: dropdownAssetItem.assetName,
+            // `tokenDecimals` would result in previously selected asset value,
+            // so we have to make a lookup for new decimals value
+            quantity: parseTokenAmount(
+              fieldValue,
+              tokensMetadata[`${dropdownAssetItem.policyId}${dropdownAssetItem.assetName}`]
```

Comment:
I'd rather centralize the logic to transform token id/asset name to "subjects" so TS can keep track whether the the string is being formed properly - related to the previous comment I had regarding a branded type for "subject"

Reply:
Refactored